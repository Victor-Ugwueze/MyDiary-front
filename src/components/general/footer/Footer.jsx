// react libraries
import React from 'react';

const Footer = () => (
  <footer>
      <p>Copywright &copy; My Dairy 2018</p>
  </footer>
);

export default Footer;
