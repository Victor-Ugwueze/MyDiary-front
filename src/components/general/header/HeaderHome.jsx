// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const HeaderHome = props => (
      <header className="home-header">
          <img className="loading-indicator" src="/images/icons/loading_spinner3.gif" />
            <div className="container-fluid">
                <div className="brand">
                    <div>
                            My Dairy
                    </div>
                    <span className="icon-bar hide" id="menu-icon"></span>
                </div>
                <nav>
                    <ul className="navbar">
                        <li className="nav-link">
                            <a href="index.html">
                                Home
                            </a>
                        </li>
                        <li className="nav-link">
                            <a href="#"
                                onClick={props.openModal}
                                data-target="sign-up-login"
                                data-toggle="login"
                            >
                                login
                            </a>
                        </li>
                        <li className="nav-link" >
                            <a href="#"
                                data-toggle="signup"
                                onClick={props.openModal}
                                data-target="sign-up-login">
                                Sign up for Free
                            </a>
                        </li>
                    </ul>
                </nav>
            </div>
            <div id="intro">
                <div id="home-text">
                    <h1>Welcome to My Diary</h1>
                    <p>A place to pen down all your toughts</p>
                    <a href="#how-it-works" className="btn btn-primary how-it-works-btn">How It Works</a>
                </div>

            </div>
        </header>
);

HeaderHome.propTypes = {
  openModal: PropTypes.func,
};

export default HeaderHome;
