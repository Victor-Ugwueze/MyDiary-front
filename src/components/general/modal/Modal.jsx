// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const Modal = props => (
  <div id='sign-up-login' className={`modal ${props.isModalOpen ? 'show' : 'hide'} col-2`}>
    <div className='modal-content'>
      <div className='modal-body'>{props.children}</div>
    </div>
  </div>
);

export default Modal;

Modal.propTypes = {
  isModalOpen: PropTypes.bool,
  children: PropTypes.array,
};
