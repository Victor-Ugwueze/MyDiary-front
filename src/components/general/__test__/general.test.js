// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import Footer from '../footer/Footer';
import HeaderHome from '../header/HeaderHome';
import Modal from '../modal/Modal';


configure({
  adapter: new Adapter(),
});

describe('Renders modals', () => {
  it('should render the footer component', () => {
    const component = shallow(
      <Footer />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the header component', () => {
    const component = shallow(
      <HeaderHome />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the reusable modal component', () => {
    const component = shallow(
      <Modal isModalOpen={true}/>
    );
    expect(component).toMatchSnapshot();
  });
});
