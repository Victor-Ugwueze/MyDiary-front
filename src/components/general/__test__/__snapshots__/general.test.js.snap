// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders modals should render the footer component 1`] = `
<footer>
  <p>
    Copywright Â© My Dairy 2018
  </p>
</footer>
`;

exports[`Renders modals should render the header component 1`] = `
<header
  className="home-header"
>
  <img
    className="loading-indicator"
    src="/images/icons/loading_spinner3.gif"
  />
  <div
    className="container-fluid"
  >
    <div
      className="brand"
    >
      <div>
        My Dairy
      </div>
      <span
        className="icon-bar hide"
        id="menu-icon"
      />
    </div>
    <nav>
      <ul
        className="navbar"
      >
        <li
          className="nav-link"
        >
          <a
            href="index.html"
          >
            Home
          </a>
        </li>
        <li
          className="nav-link"
        >
          <a
            data-target="sign-up-login"
            data-toggle="login"
          >
            login
          </a>
        </li>
        <li
          className="nav-link"
        >
          <a
            data-target="sign-up-login"
            data-toggle="signup"
            href="#"
          >
            Sign up for Free
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    id="intro"
  >
    <div
      id="home-text"
    >
      <h1>
        Welcome to My Diary
      </h1>
      <p>
        A place to pen down all your toughts
      </p>
      <a
        className="btn btn-primary how-it-works-btn"
        href="#how-it-works"
      >
        How It Works
      </a>
    </div>
  </div>
</header>
`;

exports[`Renders modals should render the reusable modal component 1`] = `
<div
  className="modal show col-2"
  id="sign-up-login"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-body"
    />
  </div>
</div>
`;
